tiledb/sm/config/config.cc

```diff
@@ -916,39 +926,60 @@ const char* Config::get_from_config(
   return *found ? it->second.c_str() : "";
 }
 
-const char* Config::get_from_config_or_env(
+const char* Config::get_from_config_or_fallback(
     const std::string& param, bool* found) const {
-  // First let's see if the User has set the parameter
-  // If it is not a user set parameter it might be a default value if found in
-  // the config
+  // First check if the user has set the parameter
+  // If not, it may be a default value (if found in the config)
   bool user_set_parameter = set_params_.find(param) != set_params_.end();
 
-  // First check config
+  // [1. user-set config parameters]
   bool found_config;
   const char* value_config = get_from_config(param, &found_config);
-  // If its a user set parameter from the config return it
   if (found_config && user_set_parameter) {
     *found = found_config;
     return value_config;
   }
 
-  // Check if default for this parameter should be ignored based on
-  // environment variables
+  // Check if param default should be ignored based on environment variables
   if (ignore_default_via_env(param)) {
     *found = true;
     return "";
   }
 
-  // Check env if not found in config or if it was found in the config but is
-  // a default value
+  // [2. env variables]
   const char* value_env = get_from_env(param, found);
   if (*found)
     return value_env;
 
-  // At this point the value was not found to be user set in the config or an
-  // environmental variable so return any default value from the config or
-  // indicate it was not found
+  // [3. profiles] -- only for rest.* params
+  if (param.rfind("rest.", 0) == 0) {
```

Comment:
My gut says that this is better expressed as something like `static bool RestProfile::canHaveParameter(std::string_view param)`.  This way we encapsulate all of the profile-ness in the profile class.

Reply:
Good idea!